packages/app/pages/Api/agent.tsx

```diff
@@ -0,0 +1,145 @@
+      <div>agentCommitments: {concatArray(props.agent.agentCommitments)}</div>
+      <div>agentRelationships: {concatArray(props.agent.agentRelationships)}</div>
+      <div>agentRoles: {concatArray(props.agent.agentRoles)}</div>
+      <div>agentRecipess: {concatArray(props.agent.agentRecipes)}</div>
+      <br/>
+    </div>
+  );
+};
+
+function concatArray(array: JSON[]) {
+  var retValue = "";
+  console.log("Array: ", array, "\nIndex 0: ", array[0]);
+  if (array[0] == null) {
+    return "none"
+  }
+  retValue = array[0].id;
+  for (var i = 1; i < array.length; i++) {
+    retValue += ", ";
+    retValue += array[i].id;
+  }
+  return retValue
+}
+
+const AgentField = (props) => {
+  return(
+    <div>
+      <form onSubmit={props.onSubmitAction}>
+        Enter an Id: <input type="text" name="value" onChange={props.setAgent}/>
+        <input type="submit" value="query"/>
+      </form>
+    </div>
+  );
+};
+
+export const GetMyAgent = getMyAgent(({ agent, loading, error}) => {
+  return (
+    loading ? <strong>Loading...</strong> : (
+      error ? <p style={{color: "#F00"}}>API error</p> : (
+        <div>
+          <div>
+            {<Agent agent={agent}/>}
+          </div>
+        </div>
+      )
+    )
+  );
+});
+
+export const GetAllAgents = getAllAgents(({ agent, loading, error}) => {
+  return (
+    loading ? <strong>Loading...</strong> : (
+      error ? <p style={{color: "#F00"}}>API error</p> : (
+        <div>
+          <div>
+            {concatArray(agent)}
+          </div>
+        </div>
+      )
+    )
+  );
+});
+
+export const GetSingleAgent = getAgentById(({ agent, loading, error }) => {
+    return (
+      loading ? <strong>Loading...</strong> : (
+        error ? <p style={{color: "#F00"}}>API error</p> : (
+          <div>
+            <div>
+              <Agent agent={agent}/>
+            </div>
+          </div>
+        )
+      )
+    );
+  }
+);
+
+class App extends React.Component {
+
+  state = {
+    getOneAgentId: null,
+    setOneAgentId: null
+  };
+
+  // Runs every time the input field changes
+  getAgentById = (event) => {
+    this.setState({setOneAgentId: parseInt(event.target.value, 10)});
+  };
+
+  // Runs when "submit" is selected
+  stopRefresh = (event) => {
+    // Sets the value to query to the current value of the input field
+    this.setState({getOneAgentId: this.state.setOneAgentId});
+    event.preventDefault();
+  };
+
+  render() {
+    const {getOneAgentId} = this.state;
+    return (
+      <div>
+        <h2>My Agent: </h2>
+        <br/>
+        <GetMyAgent/>
+        <br/>
+        <h2>All Agents: </h2>
+        <br/>
+        <GetAllAgents/>
+        <br/>
+        <h2>Get an Agent By Id: </h2>
+        <br/>
+        <AgentField setAgent={this.getAgentById} onSubmitAction={this.stopRefresh}/>
+        {getOneAgentId ? <GetSingleAgent agentId={getOneAgentId}/> : <div>Enter a value</div>}
```

Comment:
Just want to say again that this is a really good idea for how to render the results and run the query at the same time. I am definitely using this in my queries.

Reply:
Thanks. I think this highlights the clean-ness of good react